comparisons.py

```diff
@@ -1,32 +1,80 @@
-from abc import ABC, abstractmethod
-from typing import Set, Tuple
-from .models import Network, Node, Span
+from dataclasses import dataclass
+from typing import ClassVar, Generic, Optional, Set, Tuple, TypeVar
 
+from .models import FeatureType, Network, Node, Span
 
-class AbstractNetworkComparison(ABC):
-    def __init__(self, networkA: Network, networkB: Network):
-        self.networkA = networkA
-        self.networkB = networkB
 
-    @abstractmethod
-    def findPotentialSameNodes(self) -> Set[Tuple[Node, Node]]: ...
+@dataclass(frozen=True)
+class ComparisonReason:
```

Comment:
The way I've been thinking of it I don't think we need to track a reason for comparision (ie. we'll always compare features that are geographically close together) - only a reason for merging them (ie. that they were manually merged, or automatically due to a high similarity score on xyz fields)